### DISCUSSION TITLE
[CSV Export] How to export an object with childs and child's variants?

### DISCUSSION THREAD

**Discussion by pento-thal** [pimcore/pimcore#9389] [Q&A / Support] [answered]
Hi, I'm trying to manage the export of my catalogue but still don't understand how:

1. Let's say I've an object (A),  I would like to export all its childs (A1, A2, A3) and their variants (A1.1, A1.2, A1.3) too.
![Immagine 2021-06-09 171226](https://user-images.githubusercontent.com/83424908/121406625-ae1a4c80-c95e-11eb-8a71-6a46181aa53f.png)

2. Let's say I've an object (A), created from a class (B) with a "one-to-many" attribute (z) linked to another object (C), created from a class (D). 
The goal is to obtain a CSV export of the object (A) showing all the object (C) attributes. These are the involved classes:
![image](https://user-images.githubusercontent.com/83424908/121357284-f96a3600-c931-11eb-8820-3bbfec341ba2.png) ![image](https://user-images.githubusercontent.com/83424908/121357525-2ae30180-c932-11eb-85fc-396e54a31ea8.png)
Does anybody knows how to do that?

**Comment by weisswurstkanone** [comment]
did you have a look at the transformators?  https://pimcore.com/docs/pimcore/current/User_Documentation/DataObjects/Grid_Configuration_Operators/index.html :-)

**Reply by pento-thal** [reply]
Yes I tryed with this column configuration to get the "gender" field:
![image](https://user-images.githubusercontent.com/83424908/121373586-7d76ea80-c93f-11eb-905b-2a0f552f1cc3.png)
![image](https://user-images.githubusercontent.com/83424908/121374675-5ff65080-c940-11eb-8f32-d52d5fc5cd3b.png)

But nothing happenened.

**Reply by pento-thal** [reply]
@weisswurstkanone suddenly (after a couple of logout-login) it finally works! Thanks a lot!

Do you have any suggestion to solve my other issue: 
by exporting the object (A) I would like to export all its child (A1, A2, A3) and their variants (A1.1, A1.2, A1.3) too.
![Immagine 2021-06-09 171226](https://user-images.githubusercontent.com/83424908/121382285-bb2b4180-c946-11eb-88f1-f65ff6227d91.png)

Do you think it is possible?

**Comment by weisswurstkanone** [comment]
maybe by nesting operators. in the worst case you could still use the phpcode operator which allows to inject your own logic.

**Reply by pento-thal** [reply]
Hi @weisswurstkanone, what do you mean with "nesting operators"? how should it works?

**Comment by pento-thal** [comment, marked as answer]
Problem solved!
It was sufficient go to object class basic configuration and mark these options;
![image](https://user-images.githubusercontent.com/83424908/121492458-30465780-c9d7-11eb-83fc-b2cec4a40cc4.png)
